src/main/java/com/redhat/rcm/offliner/Main.java

```diff
@@ -300,6 +308,24 @@ private ArtifactListReader getArtifactListReader( File file )
             try
             {
                 final File target = new File( opts.getDownloads(), path );
+
+                if ( target.exists() )
+                {
+                    if ( null == checksums || checksums.isEmpty() || !checksums.containsKey( path ) )
+                    {
+                        return DownloadResult.avoid( path, true );
+                    }
+
+                    byte[] b = FileUtils.readFileToByteArray( target );
+                    String original = checksums.get( path );
+                    String current = sha256Hex( b );
```

Comment:
For the pom input case, I make a null checksum map as param to pass instead of a checksum map <paths, nulls>, you could check here: https://github.com/release-engineering/offliner/blob/master/src/main/java/com/redhat/rcm/offliner/alist/PomArtifactListReader.java#L211
So for this original check, will not have the NPE sistuation caused by null input, because they has been dealed in the if ( null == checksums || checksums.isEmpty() ...), while to make sure the code is on the safe side, I'll add the original null check based on your comment, John, thanks a lot!